blockchain/chain.go

```diff
@@ -1069,6 +1072,13 @@ func (b *BlockChain) reorganizeChainInternal(target *blockNode) error {
 			return err
 		}
 
+		// Add any spend journal consumer dependencies for the block
+		// to be disconnected.
+		err = b.spendPruner.AddSpendConsumerDeps(block.Hash())
+		if err != nil {
+			return err
+		}
```

Comment:
Could this be handled internally within `MaybePruneSpendData`? (i.e., `MaybePruneSpendData` would first add any dependencies and then only remove the spend journal entry if there are no dependencies).  I think it would simplify the flow unless there is a reason why they need to be separate.

Reply:
It could but it'd be less readable, I think it's beneficial to distinguish between the two processes, don't think there is much gained combining them under `MaybePruneSpendData`.

Reply:
You could still have a separate function for adding dependencies within the `SpendJournalPruner`, but I don't see why `BlockChain` needs to know about that.  Generally, fewer entry points between components leads to better encapsulation and more readable code.  Also, if you do decide to keep them separate, it seems that `AddSpendConsumerDeps` can be called immediately before `MaybePruneSpendData` rather than having them in two separate places, correct?

Reply:
Yes you're correct, updating.